src/main/java/bio/terra/service/snapshotbuilder/query/TableVariable.java

```diff
@@ -126,5 +127,9 @@ public FieldVariable getJoinFieldOnParent() {
     public boolean isLeftJoin() {
       return this.isLeftJoin;
     }
+
+    public T build() {
+      throw new InternalServerErrorException("Should not use Table Variable as is");
```

Comment:
I was wondering if we could use the generics to make this do the right thing. `new T` doesn't work. I think we could do something where we added a `TableVariable.new(constructor params)` static method, and then overrode it for each of the subtypes. I am not sure that buys us much though.